components/CodeMirror.vue

```diff
@@ -82,38 +71,15 @@ export default {
     onChanges(cm, changes) {
       this.$emit('onChanges', cm, changes);
     },
-
-    fit() {
-      if ( !this.autoResize ) {
-        return;
-      }
-
-      const container = $(this.$refs.cm.$el);
-
-      if ( !container || !container.length ) {
-        return;
-      }
-
-      const offset = container.offset();
-
-      if ( !offset ) {
-        return;
-      }
-
-      const desired = $(window).innerHeight() - offset.top - this.footerSpace;
-
-      container.css('height', `${ Math.max(this.minHeight, desired) }px`);
-    },
```

Comment:
The CSS auto height you have now is not equivalent to this.

The desired behavior is that:
- The editor expands to take up the available height of the main body (even if there's only a couple lines in it) so that it is not constantly changing size and the position of the save/cancel buttons as you type.
- There's a minHeight that also can apply to ensure a reasonable amount is viewable.
- If there are too many lines to show, the editor contents scrolls (not the actual page body).
- It leaves space for the create/cancel buttons so they are always visible at the bottom.
- The header ("Create Thing") is indirectly fixed, because the main body isn't scrolling* since the editor sizes itself to fit as above.

*: That this is not currently true because the space left at the bottom is failing allowing for the page footer, so the body scrolls slightly and that's not good.

![image](https://user-images.githubusercontent.com/753917/78710944-083b0000-78cb-11ea-9ad5-9e0cdb9a8891.png)

Reply:
You're correct that it's not equivalent. Based on the prop name I do think it's far more consistent with the code-mirror intent and documentation. https://codemirror.net/demo/resize.html

The original code also did two things which made the component non-composable.
1. If autoresize was false and there wasn't content the editor wasn't displayed at all.
2. If autoresize was set the editor would overlap with content below.

If possible I think it would make more sense to move the described behavior into ResourceYaml. Would that be okay with you?

Reply:
I'm aware of what their option does (and this is using it) but we don't have a use-case for its behavior.  The desired behaviors (which I confirmed with @tfiduccia) are:

If the editor is the entirety of the form or close then it should be at the bottom, fill the screen (but not more) and leave room for the submit actions (which is the current "autoResize".  You can call it fit to screen or something instead if you want?  And it could be off by default).  I think this includes Gatekeeper's use-cases.

If the editor is for a field in the middle of a form then it should have a bounded min/max size, like `<TextAreaAutoGrow>`s  do.  (I don't know that we have any uses of this yet)  (You can have max=0 be the native Infinity behavior, even).

I don't think the sizing behavior's usage is limited to ResourceYaml.  e.g. the GatekeeperConfig page has a CodeMiirror instance that should do the same.  If we add a text or JSON editor for something  they'd probably work the same.  

The lines between generic `<CodeMirror>`, a `<YamlEditor>`, and a `<ResourceYaml>` seems a little unclear.

`<CodeMirror>`: Pretty vanilla CodeMirror, with reasonable defaults and sizing options (maybe it shouldn't default to yaml mode to be more clear)?
`<YamlEditor>`: Adds linting and diff (but the diff isn't really yaml-specific)?
`<ResourceYaml>`: Adds special folding and edit behavior; almost a complete "page", handles the actual update/create call?

Reply:
I've completely removed autoResize, footerSpace and minHeight. Instead I refactored using css and I think it accounts for all of the requirements except for

> The header ("Create Thing") is indirectly fixed, because the main body isn't scrolling* since the editor sizes itself to fit as above.

That requirement conflicts with the min-height requirement. Eventually the body will scroll if the viewport is small enough.

Let me know if you're okay with this new solution.

Reply:
Yes min-height can make the page scroll for small windows.. have to do something like that or you could end up with 0px of editor available.